my-app/src/frontend/reducers/auth-reducer.js

```diff
@@ -0,0 +1,32 @@
+function authReducer(state, action) {
+  switch (action.type) {
+    case "FIRST_NAME":
+      return { ...state, user: { ...state.user, firstName: action.payload } };
+
+    case "LAST_NAME":
+      return { ...state, user: { ...state.user, lastName: action.payload } };
+
+    case "EMAIL":
+      return { ...state, user: { ...state.user, email: action.payload } };
+
+    case "PASSWORD":
+      return { ...state, user: { ...state.user, password: action.payload } };
+
+    case "CONFIRM_PASSWORD":
+      return {
+        ...state,
+        user: { ...state.user, confirmPassword: action.payload },
+      };
```

Comment:
A change of approach is required here.
All this can be and should be done in useState on the respective page and not in context.

Reply:
Thanks for your suggestion but I like this approach better.